src/Hocon.API.Tests/HoconAPISpec.ApproveCore.approved.txt

```diff
@@ -1,122 +1,500 @@
+        public System.Collections.Immutable.IImmutableList<Hocon.HoconElement> RemoveAt(int index) { }
+        public System.Collections.Immutable.IImmutableList<Hocon.HoconElement> RemoveRange(System.Collections.Generic.IEnumerable<Hocon.HoconElement> elements, System.Collections.Generic.IEqualityComparer<Hocon.HoconElement> equalityComparer) { }
+        public System.Collections.Immutable.IImmutableList<Hocon.HoconElement> RemoveRange(int index, int count) { }
+        public System.Collections.Immutable.IImmutableList<Hocon.HoconElement> Replace(Hocon.HoconElement oldValue, Hocon.HoconElement newValue, System.Collections.Generic.IEqualityComparer<Hocon.HoconElement> equalityComparer) { }
+        public System.Collections.Immutable.IImmutableList<Hocon.HoconElement> SetItem(int index, Hocon.HoconElement element) { }
+        public override string ToString(int indent, int indentSize) { }
+        public override bool TryGetBigIntegerList(out System.Collections.Generic.IList<System.Numerics.BigInteger> result) { }
+        public override bool TryGetBooleanList(out System.Collections.Generic.IList<bool> result) { }
+        public override bool TryGetByteList(out System.Collections.Generic.IList<byte> result) { }
+        public override bool TryGetCharList(out System.Collections.Generic.IList<char> result) { }
+        public override bool TryGetDecimalList(out System.Collections.Generic.IList<decimal> result) { }
+        public override bool TryGetDoubleList(out System.Collections.Generic.IList<double> result) { }
+        public override bool TryGetFloatList(out System.Collections.Generic.IList<float> result) { }
+        public override bool TryGetIntList(out System.Collections.Generic.IList<int> result) { }
+        public override bool TryGetLongList(out System.Collections.Generic.IList<long> result) { }
+        public override bool TryGetObjectList(out System.Collections.Generic.IList<Hocon.HoconObject> result) { }
+        public override bool TryGetSByteList(out System.Collections.Generic.IList<sbyte> result) { }
+        public override bool TryGetShortList(out System.Collections.Generic.IList<short> result) { }
+        public override bool TryGetStringList(out System.Collections.Generic.IList<string> result) { }
+        public override bool TryGetTimeSpanList(out System.Collections.Generic.IList<System.TimeSpan> result, bool allowInfinite = True) { }
+        public override bool TryGetUIntList(out System.Collections.Generic.IList<uint> result) { }
+        public override bool TryGetULongList(out System.Collections.Generic.IList<ulong> result) { }
+        public override bool TryGetUShortList(out System.Collections.Generic.IList<ushort> result) { }
     }
-    public sealed class HoconBool : Hocon.HoconLiteral
+    public sealed class HoconArrayBuilder : System.Collections.Generic.List<Hocon.HoconElement>
```

Comment:
Like the idea of using the builder pattern to build up the HOCON, but we need to avoid inheriting directly from mutable base classes

Reply:
There is no danger, the builder pattern is only used for building the hocon classes, all the returned built classes are immutable.